finality-aleph/src/new_network/split.rs

```diff
@@ -0,0 +1,231 @@
+use crate::new_network::{ComponentNetwork, Data, ReceiverComponent, SendError, SenderComponent};
+use aleph_bft::Recipient;
+use codec::{Decode, Encode};
+use futures::channel::mpsc;
+use log::warn;
+use std::{marker::PhantomData, sync::Arc};
+use tokio::{
+    sync::Mutex,
+    time::{interval, Duration},
+};
+
+const MAXIMUM_RETRY_MS: u64 = 500;
+
+/// Used for routing data through split networks.
+#[derive(Clone, Encode, Decode)]
+pub enum Split<D1: Data, D2: Data> {
```

Comment:
Consider renaming `D1, D2` to smth like `LData, RData`

Reply:
Renamed to `LeftData` and `RightData`.